sql/src/main/java/org/apache/druid/sql/avatica/ErrorHandler.java

```diff
@@ -47,58 +49,67 @@
   }
 
   /**
-   * Logs a throwable at the error level and sanitizes the throwable if applicable. Will return
-   * the sanitized or original throwable.
+   * Logs any throwable and string format message with args at the error level.
    *
-   * @param error A throwable that will be logged then sanitized
-   * @param <T>
-   * @return The sanitized throwable
+   * @param error   the Throwable to be logged
+   * @param message the message to be logged. Can be in string format structure
+   * @param format  the format arguments for the format message string
+   * @param <T>     any type that extends throwable
+   * @return the original Throwable
    */
-  public <T extends Throwable> RuntimeException logFailureAndSanitize(T error)
+  public static <T extends Throwable> T logFailure(T error, String message, Object... format)
   {
-    return logFailureAndSanitize(error, error.getMessage());
+    LOG.error(error, message, format);
+    return error;
   }
 
   /**
-   * Logs an error message at the error level and sanitizes the throwable if applicable. Will return
-   * the sanitized or original throwable.
+   * Logs any throwable at the error level with the throwables message.
    *
-   * @param error   the throwable that will be sanitized
-   * @param message the error string formate message to be logged
-   * @param format  the format args for the message
-   * @param <T>
-   * @return A sanitized version of the throwable if applicable otherwise the original throwable
+   * @param error the throwable to be logged
+   * @param <T>   any type that extends throwable
+   * @return the original Throwable
    */
-  public <T extends Throwable> RuntimeException logFailureAndSanitize(T error, String message, Object... format)
+  public static <T extends Throwable> T logFailure(T error)
   {
-    logFailure(error, message, format);
-    return sanitize(error);
-  }
-
-  public <T extends Throwable> T logFailure(T error, String message, Object... format) {
-    LOG.error(error, message, format);
-    return error;
-  }
-
-  public <T extends Throwable> T logFailure(T error) {
     logFailure(error, error.getMessage());
     return error;
   }
 
+  /**
+   * Sanitizes a Throwable. If it's a runtime exception and it's cause is sanitizable it will sanitize that cause and
+   * return that cause as a sanitized RuntimeException.  This will do best effort to keep original exception type. If
+   * it's a checked exception that will be turned into a QueryInterruptedException.
+   * <p>
+   * This was created to sanitize some exceptions that do not need to be logged.
+   *
+   * @param error The Throwable to be sanitized
+   * @param <T>   Any class that extends Throwable
+   * @return The sanitized Throwable
+   */
   public <T extends Throwable> RuntimeException sanitize(T error)
   {
-    if (error instanceof QueryInterruptedException) {
```

Comment:
Why is the check for QueryInterruptedException removed?

Reply:
Because `QueryException` is all that is needed.

Reply:
and pattern of
```java
if (error instanceof Class) {
    return (Class) errorResponseTransformStrategy.transformIfNeeded((Class) error)
}
```
doesn't work well if using `QueryInterupptedException` since the Sanitize method that relies on is defined in `QueryException`and that returns a `QueryException` leading to a `java.lang.ClassCastException`